packages/scm/src/browser/scm-provider.ts

```diff
@@ -69,11 +69,28 @@ export interface ScmCommand {
 }
 
 export interface ScmCommit {
-    id: string,  // eg Git sha or Mercurial revision number
-    summary: string,
-    authorName: string,
-    authorEmail: string,
-    authorDateRelative: string
+    id: string;  // eg Git sha or Mercurial revision number
+    commitDetailUri: URI;
+    summary: string;
+    messageBody?: string;
+    authorName: string;
+    authorEmail: string;
+    /**
+     * The date of the commit in ISO format.
+     */
+    authorTimestamp: string;
+    authorDateRelative: string;
+    fileChanges: ScmFileChange[];
+    commitDetailOptions: {};
```

Comment:
It is not used in scm module? Could we get rid of unused properties and move them downstream extensions which are using it?

Reply:
You are correct this is not read in the scm package.  However it is read in both the git and the scm-extra packages.  In scm-extra package it is passed through to the 'open' method, so is used in git-commit-detail-open-handler.ts, doOpen.  I don't see how it can be moved down to the git and scm-extra packages because it must be the same value in both cases.

Reply:
Cannot scm-extra define some additional types with `is` function for down-casting and git extension making use of them?